testsuite/generate-format/src/consensus.rs

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use libra_crypto::{
+    ed25519::{Ed25519PrivateKey, Ed25519PublicKey},
+    hash::{CryptoHasher, TestOnlyHasher},
+    traits::{SigningKey, Uniform},
+};
+use rand::{rngs::StdRng, SeedableRng};
+use serde::{Deserialize, Serialize};
+use serde_reflection::{Registry, Result, Samples, Tracer, TracerConfig};
+
+/// Return a relative path to start tracking changes in commits.
+pub fn output_file() -> Option<&'static str> {
+    Some("tests/staged/consensus.yaml")
+}
+
+/// Record sample values for crypto types used by consensus.
+fn trace_crypto_values(tracer: &mut Tracer, samples: &mut Samples) -> Result<()> {
+    let mut hasher = TestOnlyHasher::default();
+    hasher.write(b"Test message");
+    let hashed_message = hasher.finish();
+
+    let mut rng: StdRng = SeedableRng::from_seed([0; 32]);
+    let private_key = Ed25519PrivateKey::generate(&mut rng);
+    let public_key: Ed25519PublicKey = (&private_key).into();
+    let signature = private_key.sign_message(&hashed_message);
+
+    tracer.trace_value(samples, &public_key)?;
+    tracer.trace_value(samples, &signature)?;
+    Ok(())
+}
+
+/// Placeholder type.
+#[derive(Serialize, Deserialize, Eq, PartialEq, Debug, Default, Clone)]
+struct Payload;
```

Comment:
Is there a reason we can't use the Payload type that is actually used in Consensus?

Reply:
We could but this would pull all the dependencies in `consensus.yaml`.